CON-3950 General chmn discounts

AntonioKoB opened the pull request.

MikeAdamsMews reviewed (changes requested):
Hi @AntonioKoB, I have a few general comments:

1.) We need a Changelog entry to say what the change is, e.g. "[CHM: Update prices](../channel-manager-operations/inventory.md#update-prices) extended with dependent rates."

2.) The spelling should be "dependent" everywhere, not "dependant"

3.) I'm not keen on adding all the whitespace to the tables, we don't follow that convention anywhere else in the docs in any of the APIs and I would rather keep it consistent. I suspect it may be the software you are using to edit the Markdown that's adding this whitespace?

4.) You are right of course, the code snippets should be json not javascript, I can't even remember why we use javascript. Let's change it to json and I will make a note to research this and if it's okay I really ought to update all the code snippets everywhere.

Thanks :-)

AntonioKoB commented:
Hi @MikeAdamsMews 
Yes, I was using Rider and it was automatically adding the spaces..
I did addressed your comments.

I did change from Javascript to JSON also because Rider, as it was highlighting some IntelliSense warnings.

patrikmiksovsky95 commented on channel-manager-operations/inventory.md:
```diff
@@ -75,6 +89,24 @@ Mews always pushes both `gross` and `net` prices, the channel manager chooses wh
+| Property | Type | Contract | Description |
+|:--|:--| :-- |:--|
+| `rateCode`     | `string` | required | Dependent rate code.                   |
+| `baseRateCode` | `string` | required | Base rate code of Dependent rate code. |
```
This technically is optionaly because it can happen base rate is not conencted

patrikmiksovsky95 replied:
And I would make it as comment that it in documentation

MikeAdamsMews commented on changelog/README.md:
```diff
@@ -1,5 +1,9 @@
 
+## 3rd October 2024
+
+* [Inventory Update prices](/channel-manager-operations/inventory.md) extended with `dependent rates`.
```
As a general rule, the name of the reference should be the same as the thing that it is referencing, eg. "Update prices" for a reference to the API Operation "Update prices". However, uniquely on the Channel Manager API it's good to add either "CHM:" (Channel Manager side) or "Mews:" (Mews side) to the front, because the API is in two halves, and sometimes operations have the same name on both sides.
```suggestion
* [CHM: Update prices](/channel-manager-operations/inventory.md) extended with `dependent rates`.
```

MikeAdamsMews commented on changelog/README.md:
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 3rd October 2024
```
```suggestion
## 4th October 2024
```

MikeAdamsMews reviewed (approved):
Thanks for your changes @AntonioKoB. Just a couple of small suggestions for the Changelog entry, otherwise we're good.
